### PR TITLE
Devcon 6 permissionless perks cashback integration

### PR DESCRIPTION
**Pull request by micwallace**
Hey WindingTree team,

Here is the initial implementation of the permissionless perks cashback offer. 

Note that for the offer to show you need to set this in the .env:
`REACT_APP_STL_DEVCON=1`

And make a booking in Begota. The booking confirmation page will show the offer. Clicking on "Claim" will invoke token-negotiator, load devcon tokens & invoke the authentication process. The result of the process is sent to our server for verification.  

Please let me know if you have any questions or change requests.

### CODE REVIEW

**Comment by kostysh** [comment]
- code not typed properly
- page params accessed directly from document.location.search instead of router hook
- direct access to the process.env. in the component instead of via config
- ...

File: config-overrides.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
```
[the remark this answers is not in the document]
Comment (micwallace) [reply config-overrides.js]: No worries, done :-)

File: src/containers/booking-confirmation/DevconCashbackReward.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
+import {Client} from "@tokenscript/token-negotiator";
```
[the remark this answers is not in the document]
Comment (micwallace) [reply src/containers/booking-confirmation/DevconCashbackReward.tsx:1]: I think until after Devcon, or whenever the cashback limit is exceeded. Please check with Sunil on that

File: src/containers/booking-confirmation/DevconCashbackReward.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+export const DevconCashbackReward = () => {
+
+  const { provider } = useAppState();
+
+  const theme = useTheme();
+  const paddingContainer = theme.spacing(3);
+
+  const [modalMode, setModalMode] = useState(0);
```
[the remark this answers is not in the document]
Comment (micwallace) [reply src/containers/booking-confirmation/DevconCashbackReward.tsx]: I have made it an enum to make it more verbose

File: config-overrides.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = function override(config, env) {
+
+    if (!config.plugins) {
```
[the remark this answers is not in the document]
Comment (micwallace) [reply config-overrides.js]: Yep, accidentally left this here. Usually we need these polyfills but looks like you've taken care of it elsewhere.

**Comment by micwallace** [comment]
Hey guys, I've added most of the requested changes in the last commit. 

I'm just not sure about this:
`direct access to the process.env. in the component instead of via config`

I took a look in config.ts and couldn't see any abstraction for accessing environment vars. Please let me know how I should access, or feel free to go ahead and make the required changes.

**Comment by valera-rozuvan** [comment]
So, @micwallace , in response to your question: 

> I took a look in config.ts and couldn't see any abstraction for accessing environment vars. Please let me know how I should access, or feel free to go ahead and make the required changes.

See the recent PR https://github.com/windingtree/win-frontend/pull/202 . It has some work in the `config` logic, and also some changes in app logic which use the variables from the config.

**Comment by kostysh** [comment]
> Hey guys, I've added most of the requested changes in the last commit.
> 
> I'm just not sure about this: `direct access to the process.env. in the component instead of via config`
> 
> I took a look in config.ts and couldn't see any abstraction for accessing environment vars. Please let me know how I should access, or feel free to go ahead and make the required changes.

I am going to fix all the rest and notify you then PR will be merged and can be checked on the test.win.so

**Comment by kostysh** [comment]
I have forked and refactored your PR. My PR is here: https://github.com/windingtree/win-frontend/pull/206
This PR will be closed.

**Comment by micwallace** [comment]
Really appreciate this @kostysh will you be at Devcon?

**Comment by kostysh** [comment]
@micwallace not sure. having some troubles with my status/visas because of the war in Ukraine. I cannot leave Europe. Maybe next time

**Pull request closed without merging**